plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java

```diff
@@ -65,11 +80,81 @@ public void createControl(Composite parent) {
 		setInitialProjectName("org.xtext.example." + projectsuffix); //$NON-NLS-1$
 		super.createControl(parent);
 		createLanguageSelectionGroup((Composite) getControl());
+		creatBreeGroup((Composite) getControl());
 		createWorkingSetGroup((Composite) getControl(), selection, getWorkingSetIdents());
 		setDefaults(projectsuffix);
 		Dialog.applyDialogFont(getControl());
 	}
 
+	protected void creatBreeGroup(Composite parent) {
+
+		Group breeGroup = new Group(parent, SWT.NONE);
+		breeGroup.setFont(parent.getFont());
+		breeGroup.setText(Messages.WizardNewXtextProjectCreationPage_EEGrTitle);
+		breeGroup.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+		breeGroup.setLayout(new GridLayout(1, false));
+
+		Composite composite = new Composite(breeGroup, SWT.NONE);
+		composite.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+		composite.setLayout(new GridLayout(3, false));
+
+		Label breeLabel = new Label(composite, SWT.NONE);
+		GridData data = new GridData(GridData.FILL_HORIZONTAL);
+		data.horizontalSpan = 1;
+		breeLabel.setText(Messages.WizardNewXtextProjectCreationPage_EECombo);
+
+		breeCombo = new Combo(composite, SWT.BORDER | SWT.READ_ONLY);
+		data = new GridData(GridData.FILL_HORIZONTAL);
+		data.horizontalSpan = 1;
+		breeCombo.setLayoutData(data);
+		breeCombo.setFont(parent.getFont());
+		fillBreeCombo(breeCombo);
+
+		Listener modifyListener = new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				setPageComplete(validatePage());
+			}
+		};
+		breeCombo.addListener(SWT.Modify, modifyListener);
+		// Create button
+		exeEnvButton = new Button(composite, SWT.PUSH);
+		exeEnvButton.setLayoutData(new GridData());
+		exeEnvButton.setText(Messages.WizardNewXtextProjectCreationPage_EEButton);
+		exeEnvButton.addListener(SWT.Selection, new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				PreferencesUtil.createPreferenceDialogOn(getShell(), "org.eclipse.jdt.debug.ui.jreProfiles", //$NON-NLS-1$
+						new String[] { "org.eclipse.jdt.debug.ui.jreProfiles" }, null).open(); //$NON-NLS-1$ 
+				fillBreeCombo(breeCombo);
+			}
+		});
+	}
+
+	protected void fillBreeCombo(Combo comboToFill) {
+		Set<String> brees = Sets.newHashSet(JREContainerProvider.getDefaultBREE());
+		Set<String> availableBrees = Sets.newHashSet();
+		for (IExecutionEnvironment ee : JavaRuntime.getExecutionEnvironmentsManager().getExecutionEnvironments()) {
+			availableBrees.add(ee.getId());
+		}
+		for (JavaVersion supportedVersion : JavaVersion.values()) {
+			if (supportedVersion.isAtLeast(JavaVersion.JAVA6)) {
+				String bree = supportedVersion.getBree();
+				if (availableBrees.contains(bree))
+					brees.add(bree);
+			}
+		}
+		String[] array = brees.toArray(new String[] {});
+		Arrays.sort(array, Policy.getComparator());
```

Comment:
Will this always sort alphanumerically where we want to sort by version? Maybe not a big deal since currently both variants will yield the same result. A java 10 or smth may lead to issues here.